crates/spirv-std/macros/src/lib.rs

```diff
@@ -153,7 +160,11 @@ pub fn spirv(_attr: TokenStream, item: TokenStream) -> TokenStream {
                                 && matches!(group.stream().into_iter().next(), Some(TokenTree::Ident(ident)) if ident == "spirv")
                                 && matches!(sub_tokens.last(), Some(TokenTree::Punct(p)) if p.as_char() == '#') =>
```

Comment:
Hmm this side feels very sketchy but I guess we can't use the proc macro attribute directly on e.g. function arguments?

Reply:
This isn't my code though. But yeah, that's why we had the proc macro attribute implementation in the first place, to strip all those illegal uses of the attribute.

Also it only matches literal `#[spirv()]`. If someone does `#[spirv_std::spirv()]` we don't catch that.

Reply:
> Also it only matches literal `#[spirv()]`. If someone does `#[spirv_std::spirv()]` we don't catch that.

Right, this is annoying, but it doesn't look like we can do better for now - I was able to find a crate on `play.rust-lang.org` to test with and yeah: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=b23176b6523a4ae675fd7e65b1fb6d4f

(even on nightly there's no feature-gate suggested so I think it's just not supported at all)